src/types/magnitude.rs

```diff
@@ -71,7 +71,10 @@ impl Magnitude {
 
 impl From<BigUint> for Magnitude {
     fn from(value: BigUint) -> Self {
-        Magnitude::BigUInt(value)
+        match value.to_u64() {
+            Some(unsigned) => Magnitude::U64(unsigned),
+            None => Magnitude::BigUInt(value),
+        }
```

Comment:
May be contentious, but I think this is right. The other place to do this is at the call-site (conversion from BigDecimal to Decimal). I think this is more robust.

Reply:
This seems reasonable to me--we are making sure that the conversion is more compact, right?